libraries/AP_RPM/AP_RPM.cpp

```diff
@@ -161,14 +95,88 @@ void AP_RPM::init(void)
     }
 }
 
+void AP_RPM::convert_params(void)
+{
+    if (_params[0]._type.configured_in_storage()) {
+        // _params[0]._type will always be configured in storage after conversion is done the first time
+        return;
+    }
+
+    struct ConversionTable {
+        uint8_t old_element;
+        uint8_t new_index;
+        uint8_t instance;
+    };
+
+    const struct ConversionTable conversionTable[] = {
+            // RPM 1
+            {0, 0, 0}, // TYPE
+            {1, 1, 0}, // SCALING
+            {2, 2, 0}, // MAX
+            {3, 3, 0}, // MIN
+            {4, 4, 0}, // MIN_QUAL
+            {5, 5, 0}, // PIN
+            {6, 6, 0}, // ESC_MASK
+
+            // RPM 2
+            {10, 0, 1}, // TYPE
+            {11, 1, 1}, // SCALING
+            // MAX (Previous bug meant RPM2_MAX param was never accesible to users. No conversion required.)
+            // MIN (Previous bug meant RPM2_MIN param was never accesible to users. No conversion required.)
+            {4, 4, 1}, // MIN_QUAL (Previously the min quality of the 1st RPM instance was used for all RPM instances.)
+            {12, 5, 1}, // PIN
+            {13, 6, 1}, // ESC_MASK
+    };
+
+    char param_name[17] = {0};
+    AP_Param::ConversionInfo info;
+    info.new_name = param_name;
+
+#if APM_BUILD_TYPE(APM_BUILD_ArduPlane)
```

Comment:
maybe use find_top_level_key_by_pointer()

Reply:
Done and tested.  Confirmed working.